### DISCUSSION TITLE
Topography-based simulation

### DISCUSSION THREAD

**Discussion by hududed** [srsran/srsRAN_Project#821] [General Help]
Hi, I was wondering if we can simulate a gNB and UE connection based on topography and location to accommodate for propogation models. For example, a simple case would be gNB is placed 10m away and 10m higher than UE.  Is this possible e.g. in conjunction with matlab (or other tools)?

**Comment by pgawlowicz** [comment]
In case of ZMQ-based setup, you can always connect srsgnb and srsUE over gnuradio, where you can simulate channel impact on the signal.
Similar as we do here: https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html#multi-ue-emulation

**Reply by hududed** [reply]
Ok so the only option currently is to first determine the path loss e.g. via ray tracing model in MATLAB, then define that path loss in gnuradio?

I am wondering if there is some level of integration with srsran-matlab

**Reply by pgawlowicz** [reply]
The signal samples are sent over zmq sockets, so I believe it should be possible to receive them directly in MATLAB. However, we do not currently have such an example.

**Reply by hududed** [reply]
Thanks @pgawlowicz  I will look further into this.

Last question - Coming back to the multi UE scenario, from the console logs, can we tell if the UEs are interfering with one another e.g. drop in brate, snr, rsrp? And, is it possible to simulate propagation delays (with ZMQ and gnuradio) and determine that from the timing advance values in the logs?

**Reply by pgawlowicz** [reply]
The UEs are allocated different radio resources, so they should not interfere with each other. Yes, you can add a delay for each UE in gnuradio, which will be reflected in the TA value.

**Reply by hududed** [reply]
> The UEs are allocated different radio resources, so they should not interfere with each other.

How do you mean different radio resources? I see in the ue1.conf and ue2.conf the opc and k values are different - is this what you mean?
Is it possible to assign them the same radio resources and see any interference behaviour?

**Reply by pgawlowicz** [reply]
The UEs connect to a gNB. The MAC scheduler is responsible for allocating radio resources for their transmissions. The UEs use only the allocated resources, hence they do not interfere with each other.

**Reply by dvdgrgrtt** [reply]
Hi!
If I can add my two cents, MATLAB doesn't support ZMQ out of the box, as far as I know. There are a couple of third-party projects that try to do the integration, but I have no experience with those, so I can't provide more feedback. I'm adding a couple of links to related post in the MATLAB forum
https://www.mathworks.com/matlabcentral/answers/723138-working-with-zmq-in-matlab-for-communicating-with-eye-tracker-device
https://www.mathworks.com/matlabcentral/answers/269061-how-do-i-integrate-zeromq-library-with-matlab-i-want-my-matlab-program-to-be-a-subscriber-of-zeromq?s_tid=srchtitle_site_search_1_zeromq

Another option is to start with the GNU Radio channel models, but I don't know if there's any support for ray-tracing
https://www.gnuradio.org/doc/doxygen/page_channels.html

**Reply by pgawlowicz** [reply]
@hududed any update on this issue?

**Reply by hududed** [reply]
@dvdgrgrtt thanks I'll look into it 

@pgawlowicz sorry thought this was open ended discussion, you can close this if you like. But one last question - is there a way to configure MAC to simulate interference between multi UEs?

**Reply by pgawlowicz** [reply]
no, there is no such possibility as MAC is not dealing with the waveform signal.
